pymatgen/util/testing/aims.py

```diff
@@ -0,0 +1,106 @@
+"""Helper functions for testing AIMS IO."""
+
+from __future__ import annotations
+
+import gzip
+import json
+from glob import glob
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import numpy as np
+from monty.io import zopen
+
+from pymatgen.core import Molecule, Structure
+
+if TYPE_CHECKING:
+    from pymatgen.io.aims.sets.base import AimsInputGenerator
+
+
+def check_band(test_line: str, ref_line: str) -> bool:
+    """Check if band lines are the same."""
+    test_pts = [float(inp) for inp in test_line.split()[-9:-2]]
+    ref_pts = [float(inp) for inp in ref_line.split()[-9:-2]]
+
+    return np.allclose(test_pts, ref_pts) and test_line.split()[-2:] == ref_line.split()[-2:]
+
+
+def compare_files(test_name: str, work_dir: Path, ref_dir: Path) -> None:
+    """Compare files generated by tests with ones in reference directories."""
+    for file in glob(f"{work_dir / test_name}/*in"):
+        with open(file) as test_file:
+            test_lines = [line.strip() for line in test_file.readlines() if len(line.strip()) > 0 and line[0] != "#"]
+
+        with gzip.open(f"{ref_dir / test_name / Path(file).name}.gz", "rt") as ref_file:
+            ref_lines = [line.strip() for line in ref_file.readlines() if len(line.strip()) > 0 and line[0] != "#"]
+
+        for test_line, ref_line in zip(test_lines, ref_lines):
+            if "output" in test_line and "band" in test_line:
+                assert check_band(test_line, ref_line)
+            else:
+                assert test_line == ref_line
+
+    with open(f"{ref_dir / test_name}/parameters.json") as ref_file:
+        ref = json.load(ref_file)
+    ref.pop("species_dir", None)
+    ref_output = ref.pop("output", None)
+
+    with open(f"{work_dir / test_name}/parameters.json") as check_file:
+        check = json.load(check_file)
+
+    check.pop("species_dir", None)
+    check_output = check.pop("output", None)
+
+    assert ref == check
+
+    if check_output:
+        for ref_out, check_out in zip(ref_output, check_output):
+            if "band" in check_out:
+                assert check_band(check_out, ref_out)
+            else:
+                assert ref_out == check_out
+
+
+def comp_system(
```

Comment:
you're defining two `comp_system` functions. i think the one in `tests/io/aims/test_sets/test_static_restart_from_relax_generator.py#L12` can go and be replaced by this one?

Reply:
also, feel free to improve the doc strings on these helper methods. i added very generic ones but you have a better understanding of what they do

Reply:
The two are slightly different, but I can combine them

I'll add more complete docstrings now that I am in the office

Reply:
I made these two fixes